server/src/test/java/org/opensearch/monitor/fs/FsHealthServiceTests.java

```diff
@@ -347,7 +403,7 @@ public void force(boolean metaData) throws IOException {
 
     private static class FileSystemFsyncHungProvider extends FilterFileSystemProvider {
 
-        AtomicBoolean injectIOException = new AtomicBoolean();
+        AtomicBoolean injectIODelay = new AtomicBoolean();
```

Comment:
Th delay is achieved by using Thread.sleep. You could use latching to make this more deterministic in tests.

Reply:
The tests actually does a latch, we are mimicking a stuck IO operations which is being achieved by causing the fsync operation to go into a sleep. I have however improved the mock to make it more deterministic based on your suggestion